skorch/tests/test_lr_scheduler.py

```diff
@@ -63,6 +43,50 @@ def _lr_callback_init_policies(
             net.callbacks_
         )))
 
+    def test_raise_invalid_policy_string(self):
+        with pytest.raises(AttributeError):
+            LRScheduler('invalid_policy')
+
+    def test_raise_invalid_policy_class(self):
+        class DummyClass():
+            pass
+
+        with pytest.raises(AssertionError):
+            LRScheduler(DummyClass)
+
+    @pytest.mark.parametrize('policy, kwargs', [
+            ('LambdaLR', {'lr_lambda': (lambda x: 1e-1)}),
+            ('StepLR', {'step_size': 30}),
+            ('MultiStepLR', {'milestones': [30, 90]}),
+            ('ExponentialLR', {'gamma': 0.1}),
+            ('ReduceLROnPlateau', {}),
+            ('WarmRestartLR', {}),
+        ])
+    def test_lr_callback_steps_correctly(self, classifier_module, policy, kwargs):
+        max_epochs = 2
+        X, y = make_classification(1000, 20, n_informative=10, random_state=0)
+        X = X.astype(np.float32)
+        lr_policy = LRScheduler(policy, **kwargs)
+        net = NeuralNetClassifier(classifier_module(), max_epochs=max_epochs, batch_size=16, callbacks=[lr_policy])
+        net.fit(X, y)
+        assert lr_policy._lr_scheduler.last_epoch == max_epochs - 1
+
+    @pytest.mark.parametrize('policy, kwargs', [
+            ('CyclicLR', {}),
+        ])
+    def test_lr_callback_batch_steps_correctly(self, classifier_module, policy, kwargs):
+        num_examples = 1000
+        batch_size = 100
+        max_epochs = 2
+
+        X, y = make_classification(num_examples, 20, n_informative=10, random_state=0)
+        X = X.astype(np.float32)
```

Comment:
You can use the `classifier_data` fixture defined in `conftest.py` for both of these operations.